mathics/core/util.py

```diff
@@ -42,6 +42,29 @@ def timed(*args, **kw):
     return timed
 
 
+class TimeitContextManager:
+    """Add this as a context manager to time parts of the code.
+
+    For example:
+        with TimeitContextManger("testing my loop"):
+           for x in collection:
+               ...
+    """
+
+    def __init__(self, name: str):
+        self.name = name
+
+    def __enter__(self):
+        # print(f"{date.today()}	{method_name} starts")
+        self.ts = time.time()
+
+    def __exit__(self, exc_type, exc_value, exc_tb):
+        te = time.time()
+        elapsed = int((te - self.ts) * 1000)
```

Comment:
Maybe `float` instead of `int`?

Reply:
Ok - that makes sense and is simpler. Done in d874cf08